snapcraft/plugins/ruby.py

```diff
@@ -0,0 +1,160 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 James Beedy <jamesbeedy@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The ruby plugin is useful for ruby based parts.
+The plugin uses gem to install dependencies from a `Gemfile`.
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+Additionally, this plugin uses the following plugin-specific keywords:
+    - gems:
+      (list)
+      A list of gems to install.
+    - use-bundler
+      (boolean)
+      Use bundler to install gems from a Gemfile (defaults 'false').
+    - ruby-version:
+      (string)
+      The version of ruby you want this snap to run.
+"""
+import os
+import re
+import platform
+
+from snapcraft import BasePlugin, file_utils
+from snapcraft.internal.errors import SnapcraftEnvironmentError
+from snapcraft.sources import Tar
+
+
+class RubyPlugin(BasePlugin):
+
+    @classmethod
+    def schema(cls):
+        schema = super().schema()
+
+        schema['properties']['use-bundler'] = {
```

Comment:
I'm rethinking this. Bundler supports being pointed to another gemfile path with `--gemfile`. I wonder if we could add a `gemfile` option that defaults to None or an empty string. Then if the user wants to enable bundler, they specify what gemfile they want to use (in most cases I expect it'll simply be 'Gemfile', but by doing it this way we get bundler support and alternative gemfile support with one stone).

I also suggest adding an option to work with `--without`, I imagine lots of users will want to exclude their test and development groups. Perhaps an option called `bundler-without-groups` that accepts a list.

Reply:
You suggested it in the first place, I only came up with an alternative syntax ;-)